extensions/blocks/amazon/edit.js

```diff
@@ -1,23 +1,248 @@
 /**
  * External dependencies
  */
+import tinycolor from 'tinycolor2';
 import { __ } from '@wordpress/i18n';
-import { Component } from '@wordpress/element';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	BlockIcon,
+	ContrastChecker,
+	InspectorControls,
+	PanelColorSettings,
+} from '@wordpress/block-editor';
+import {
+	Button,
+	FormTokenField,
+	Notice,
+	PanelBody,
+	Placeholder,
+	ToggleControl,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import icon from './icon';
+import data from './dummy-data';
 import './editor.scss';
 
-class AmazonEdit extends Component {
-	/**
-	 * Write the block editor UI.
-	 *
-	 * @returns {object} The UI displayed when user edits this block.
-	 */
-	render() {
-		return <p>{ __( 'Coming soon', 'jetpack' ) }</p>;
-	}
-}
+export default function AmazonEdit( {
+	attributes: {
+		backgroundColor,
+		textColor,
+		buttonAndLinkColor,
+		asin,
+		showImage,
+		showTitle,
+		showSeller,
+		showPrice,
+		showPurchaseButton,
+	},
+	className,
+	setAttributes,
+} ) {
+	const notice = false; // TODO
+
+	const [ suggestions, setSuggestions ] = useState( [] );
+	const onInputChange = () => {
+		// TODO get suggestions from API
+		// It would be great if we didn't have to embed the ASIN like this but I think that
+		// requires changes to core
+		setSuggestions(
+			data.products.map( dataProduct => `${ dataProduct.title } (ASIN:${ dataProduct.asin })` )
+		);
+	};
+
+	const idRegex = /^(\d+)$|\(ASIN:(.+)\)$/;
+	const onChange = selectedProducts => {
+		const selectedIds = selectedProducts.map( selectedProduct => {
+			const parsed = idRegex.exec( selectedProduct );
+			const selectedId = parsed[ 1 ] || parsed[ 2 ];
+			return data.products.filter( filteredProduct => filteredProduct.asin === selectedId );
+		} );
+		setAttributes( { asin: selectedIds[ 0 ][ 0 ].asin } );
```

Comment:
It may be interesting to add a comment here to explain what you're doing here (extracting the ASIN from the suggestion and setting it as attribute)?

Reply:
I added a comment, but when we improve/replace FormTokenField this should be greatly improved